Automattic/jetpack
# Jetpack AI Image: handle enter shortcut on user custom prompt field

Closes #38102.

## Proposed changes:

* Add a `<KeyboardShortcuts />` wrapper on the user custom prompt field to deal with `enter` hits
* When the focus is inside the custom prompt field and the user hits `enter`, we trigger the image generation, unless there is no prompt typed on it (or no content on the post as well, when it's a featured image generation)
* Rename some vars on the `AiImageModal` component so they read less like a function and more like the flags they re

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Enable the beta extensions, go to the block editor and add an image block
* Click "Select Image" and then "Generate with AI" to open the image generation tool
* Confirm the UI is showing as expected, with no buggy textareas or misalignements
* Type something on the custom prompt field (example: "a giant cup of coffee") and then hit "enter"
* Confirm the generation will start
* Do the same check for the featured image generator
   * In this case, the "enter" shortcut will work when:
      * the post has content but the custom prompt is empty
      * the post is empty but the user typed a custom prompt
      * both the post and the custom prompt have content
   * It will not work when both the post and the custom prompt are empty (as expected)
